In [20]:
import pandas as pd
import numpy as np

In [21]:
comment= pd.read_csv('comment_sentiment.csv')
comment

profilePictureUrl        username  \
0      https://scontent-vie1-1.cdninstagram.com/v/t51...          x_geek   
1      https://scontent-vie1-1.cdninstagram.com/v/t51...     garytanys88   
2      https://scontent-vie1-1.cdninstagram.com/v/t51...     whitechocoz   
3      https://scontent-vie1-1.cdninstagram.com/v/t51...  travelsofmiles   
4      https://scontent-vie1-1.cdninstagram.com/v/t51...   iamamandawang   
...                                                  ...             ...   
19054  https://scontent-yyz1-1.cdninstagram.com/v/t51...         limkirk   
19055  https://scontent-yyz1-1.cdninstagram.com/v/t51...         ttzh711   
19056  https://scontent-yyz1-1.cdninstagram.com/v/t51...    bagaholicboy   
19057  https://scontent-yyz1-1.cdninstagram.com/v/t51...         jialatt   
19058  https://scontent-yyz1-1.cdninstagram.com/v/t51...         jae7iew   

                                     profileUrl  \
0              https://www.instagram.com/x_geek   
1         https://www.instagram.com/garytanys88   
2         https://www.instagram.com/whitechocoz   
3      https://www.instagram.com/travelsofmiles   
4       https://www.instagram.com/iamamandawang   
...                                         ...   
19054         https://www.instagram.com/limkirk   
19055         https://www.instagram.com/ttzh711   
19056    https://www.instagram.com/bagaholicboy   
19057         https://www.instagram.com/jialatt   
19058         https://www.instagram.com/jae7iew   

                                                 comment  likeCount  \
0                                       Yes, So Love. 🤍✨          1   
1                                                   👍👍😍😍          1   
2                                            So black! 🖤          1   
3                                                   😍😍😍😍          1   
4                                                 ❤️❤️❤️          1   
...                                                  ...        ...   
19054  @bagaholicboy customisation and coach and cute...          1   
19055      Absolutely loveeeeeeeeeeeeeeeeeeee it!!!!!!!!          1   
19056                                #coachxbagaholicboy          0   
19057  @bagaholicboy love how u use #realwomen and #r...          2   
19058                                           Cuuuuute          1   

       replyCount               commentDate     commentId      ownerId  \
0               2  2022-03-21T14:26:07.000Z  1.790000e+16      1723935   
1               2  2022-03-21T14:22:01.000Z  1.790000e+16  45861584227   
2               0  2022-03-21T15:31:52.000Z  1.800000e+16      1378663   
3               0  2022-03-21T14:24:03.000Z  1.790000e+16    188060675   
4               0  2022-03-21T14:14:02.000Z  1.800000e+16  48881809128   
...           ...                       ...           ...          ...   
19054           0  2018-04-20T02:05:50.000Z  1.790000e+16     27254323   
19055           0  2018-04-20T01:32:08.000Z  1.790000e+16      1173257   
19056           0  2018-04-20T01:28:54.000Z  1.790000e+16      1679827   
19057           0  2018-04-20T01:21:07.000Z  1.790000e+16    423622916   
19058           0  2018-04-20T01:11:39.000Z  1.790000e+16    347011714   

                      timestamp                                     query  \
0      2022-03-22T13:48:52.066Z  https://www.instagram.com/p/CbXlb2GP6sT/   
1      2022-03-22T13:48:52.066Z  https://www.instagram.com/p/CbXlb2GP6sT/   
2      2022-03-22T13:48:52.066Z  https://www.instagram.com/p/CbXlb2GP6sT/   
3      2022-03-22T13:48:52.066Z  https://www.instagram.com/p/CbXlb2GP6sT/   
4      2022-03-22T13:48:52.066Z  https://www.instagram.com/p/CbXlb2GP6sT/   
...                         ...                                       ...   
19054  2022-03-23T03:26:41.436Z  https://www.instagram.com/p/BhxevgqBuzu/   
19055  2022-03-23T03:26:41.436Z  https://www.instagram.com/p/BhxevgqBuzu/   
19056  2022-03-23T03:26:41.436Z  https://www.instagram.com/p/BhxevgqBuzu/   
1

In [22]:
tag = comment.Tags.unique()
tag

array(['ysl', 'dior', 'cartier', 'chanel', 'louisvuitton', 'balenciaga',
       'celine', 'valentino', 'goyard', 'moynat', 'prada', 'hermes',
       'alexandermcqueen', 'gucci', 'tiffanyandco', 'loewe', 'toryburch',
       'rimowa', 'ferragamo', 'bvlgari', 'floralmagic', 'furla',
       'burberry', 'coach', 'bulgari', 'longchamp', 'audemarspiguet',
       'piaget', 'rolex', 'fendi', 'rogervivier', 'bottegaveneta',
       'versace', 'faurele', 'karllagerfeld', 'nicogiani', 'tomford'],
      dtype=object)

In [23]:
comment.describe()

likeCount    replyCount     commentId       ownerId  error  \
count  19059.00000  19059.000000  1.905900e+04  1.905900e+04    0.0   
mean       0.86269      0.261451  1.795222e+16  3.381765e+09    NaN   
std        0.68305      0.780851  9.914457e+13  9.580370e+09    NaN   
min        0.00000      0.000000  1.780000e+16  2.983000e+03    NaN   
25%        0.00000      0.000000  1.790000e+16  8.707233e+06    NaN   
50%        1.00000      0.000000  1.790000e+16  1.951473e+08    NaN   
75%        1.00000      0.000000  1.800000e+16  1.530118e+09    NaN   
max       10.00000     16.000000  1.830000e+16  5.071445e+10    NaN   

        Post_Number      Positive       Neutral      Negative      Compound  
count  19059.000000  19059.000000  19059.000000  19059.000000  19059.000000  
mean     381.498557     31.407099     64.225578      4.367291      0.408511  
std      234.282308     27.596513     28.512247     14.884218      0.434503  
min        0.000000      0.000000      0.000000      0.000000     -0.990500  
25%      189.000000      0.000000     45.600000      0.000000      0.000000  
50%      381.000000     33.300000     60.900000      0.000000      0.489800  
75%      529.000000     50.000000    100.000000      0.000000      0.812300  
max     1009.000000    100.000000    100.000000    100.000000      0.998500

In [24]:
#Proportionate of different brands
proportion = comment.groupby('Tags')["comment"].count()
df_proportion = pd.DataFrame(proportion)
df_proportion["count"] = df_proportion["comment"] 
#df_proportion["tag"] = df_proportion["Tags"]
df_proportion.drop(['comment'], axis=1, inplace=True)
df_proportion["percentage"] = df_proportion["count"] / 190.59
df_proportion

count  percentage
Tags                               
alexandermcqueen     19    0.099690
audemarspiguet       20    0.104937
balenciaga          314    1.647516
bottegaveneta        47    0.246603
bulgari             119    0.624377
burberry            189    0.991657
bvlgari              71    0.372527
cartier             435    2.282386
celine              486    2.549976
chanel             5683   29.817934
coach               162    0.849992
dior               3221   16.900152
faurele             110    0.577155
fendi               499    2.618186
ferragamo           203    1.065114
floralmagic           9    0.047222
furla                38    0.199381
goyard              412    2.161708
gucci               975    5.115693
hermes             1076    5.645627
karllagerfeld        70    0.367281
loewe               117    0.613883
longchamp            54    0.283331
louisvuitton       3124   16.391206
moynat              167    0.876226
nicogiani            12    0.062962
piaget               29    0.152159
prada               370    1.941340
rimowa              255    1.337951
rogervivier          55    0.288578
rolex                98    0.514193
tiffanyandco        165    0.865733
tomford              17    0.089197
toryburch           119    0.624377
valentino           120    0.629624
versace              39    0.204628
ysl                 160    0.839498

In [25]:
sentimental = comment.groupby('Tags')["Positive", "Neutral", "Negative", "Compound"].mean()
sentimental

C:\Users\lwl_1\AppData\Local\Continuum\anaconda3\lib\site-packages\ipykernel_launcher.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  """Entry point for launching an IPython kernel.


Positive    Neutral   Negative  Compound
Tags                                                       
alexandermcqueen  39.678947  50.873684   9.447368  0.477558
audemarspiguet    25.150000  64.095000  10.755000  0.292245
balenciaga        32.204777  60.599682   7.195541  0.376606
bottegaveneta     21.587234  62.757447  15.655319  0.200430
bulgari           34.442857  61.322689   4.234454  0.444278
burberry          33.141270  63.058730   3.798942  0.429049
bvlgari           32.573239  54.446479  12.983099  0.370970
cartier           34.766207  58.910115   6.324138  0.444491
celine            29.736420  66.630453   3.633128  0.408723
chanel            33.474415  62.292504   4.232747  0.441694
coach             27.506790  69.529630   2.963580  0.305938
dior              31.828718  64.164918   4.006333  0.410863
faurele           23.968182  73.267273   2.763636  0.357561
fendi             36.774148  60.081563   3.142886  0.480629
ferragamo         34.572906  60.199015   5.228571  0.454709
floralmagic       34.477778  65.522222   0.000000  0.415533
furla             38.681579  60.263158   1.055263  0.561847
goyard            28.706553  67.775728   3.517718  0.396474
gucci             32.170667  63.354564   4.474462  0.407198
hermes            31.173048  63.659387   5.167472  0.403666
karllagerfeld     26.402857  48.340000  25.258571  0.018364
loewe             30.533333  63.926496   5.541026  0.369432
longchamp         32.727778  61.896296   5.377778  0.407074
louisvuitton      25.379001  70.993214   3.628041  0.335873
moynat            34.464671  60.151497   5.385030  0.463594
nicogiani         44.866667  49.825000   5.308333  0.429500
piaget            33.351724  61.786207   4.865517  0.466414
prada             33.344865  62.340541   4.315676  0.458262
rimowa            35.043529  61.373725   3.583922  0.468635
rogervivier       35.276364  58.718182   6.003636  0.430000
rolex             25.666327  69.752041   4.581633  0.295670
tiffanyandco      31.442424  63.991515   4.567273  0.417196
tomford           23.735294  75.470588   0.794118  0.309776
toryburch         28.478151  68.213445   3.308403  0.330439
valentino         39.550000  56.605833   3.844167  0.536292
versace           27.446154  54.561538  17.989744  0.254600
ysl               37.263750  58.722500   4.015000  0.470644

In [26]:
sentimental["count"] = df_proportion["count"]
sentimental["percentage"] = df_proportion["percentage"]
sentimental.sort_values("percentage", ascending=False)

Positive    Neutral   Negative  Compound  count  percentage
Tags                                                                          
chanel            33.474415  62.292504   4.232747  0.441694   5683   29.817934
dior              31.828718  64.164918   4.006333  0.410863   3221   16.900152
louisvuitton      25.379001  70.993214   3.628041  0.335873   3124   16.391206
hermes            31.173048  63.659387   5.167472  0.403666   1076    5.645627
gucci             32.170667  63.354564   4.474462  0.407198    975    5.115693
fendi             36.774148  60.081563   3.142886  0.480629    499    2.618186
celine            29.736420  66.630453   3.633128  0.408723    486    2.549976
cartier           34.766207  58.910115   6.324138  0.444491    435    2.282386
goyard            28.706553  67.775728   3.517718  0.396474    412    2.161708
prada             33.344865  62.340541   4.315676  0.458262    370    1.941340
balenciaga        32.204777  60.599682   7.195541  0.376606    314    1.647516
rimowa            35.043529  61.373725   3.583922  0.468635    255    1.337951
ferragamo         34.572906  60.199015   5.228571  0.454709    203    1.065114
burberry          33.141270  63.058730   3.798942  0.429049    189    0.991657
moynat            34.464671  60.151497   5.385030  0.463594    167    0.876226
tiffanyandco      31.442424  63.991515   4.567273  0.417196    165    0.865733
coach             27.506790  69.529630   2.963580  0.305938    162    0.849992
ysl               37.263750  58.722500   4.015000  0.470644    160    0.839498
valentino         39.550000  56.605833   3.844167  0.536292    120    0.629624
toryburch         28.478151  68.213445   3.308403  0.330439    119    0.624377
bulgari           34.442857  61.322689   4.234454  0.444278    119    0.624377
loewe             30.533333  63.926496   5.541026  0.369432    117    0.613883
faurele           23.968182  73.267273   2.763636  0.357561    110    0.577155
rolex             25.666327  69.752041   4.581633  0.295670     98    0.514193
bvlgari           32.573239  54.446479  12.983099  0.370970     71    0.372527
karllagerfeld     26.402857  48.340000  25.258571  0.018364     70    0.367281
rogervivier       35.276364  58.718182   6.003636  0.430000     55    0.288578
longchamp         32.727778  61.896296   5.377778  0.407074     54    0.283331
bottegaveneta     21.587234  62.757447  15.655319  0.200430     47    0.246603
versace           27.446154  54.561538  17.989744  0.254600     39    0.204628
furla             38.681579  60.263158   1.055263  0.561847     38    0.199381
piaget            33.351724  61.786207   4.865517  0.466414     29    0.152159
audemarspiguet    25.150000  64.095000  10.755000  0.292245     20    0.104937
alexandermcqueen  39.678947  50.873684   9.447368  0.477558     19    0.099690
tomford           23.735294  75.470588   0.794118  0.309776     17    0.089197
nicogiani         44.866667  49.825000   5.308333  0.429500     12    0.062962
floralmagic       34.477778  65.522222   0.000000  0.415533      9    0.047222

In [80]:
import re
import string
import nltk

import pandas as pd

from collections import Counter
from nltk.tokenize import word_tokenize
from nltk.corpus import stopwords

nltk.download('punkt')
nltk.download('stopwords')

STOP_WORDS = stopwords.words()

# removing the emojies
# https://www.kaggle.com/alankritamishra/covid-19-tweet-sentiment-analysis#Sentiment-analysis
EMOJI_PATTERN = re.compile("["
                           u"\U0001F600-\U0001F64F"  # emoticons
                           u"\U0001F300-\U0001F5FF"  # symbols & pictographs
                           u"\U0001F680-\U0001F6FF"  # transport & map symbols
                           u"\U0001F1E0-\U0001F1FF"  # flags (iOS)
                           u"\U00002702-\U000027B0"
                           u"\U000024C2-\U0001F251"
                           "]+", flags=re.UNICODE)


def cleaning(text):
    """
    Convert to lowercase.
    Rremove URL links, special characters and punctuation.
    Tokenize and remove stop words.
    """
    text = text.lower()
    text = re.sub('https?://\S+|www\.\S+', '', text)
    text = re.sub('<.*?>+', '', text)
    text = re.sub('[%s]' % re.escape(string.punctuation), '', text)
    text = re.sub('\n', '', text)
    text = re.sub('[’“”…]', '', text)
    text = re.sub('[🥰ðÿ˜ðÿ❤œ»¨]', '', text)

    text = EMOJI_PATTERN.sub(r'', text)

    # removing the stop-words
    text_tokens = word_tokenize(text)
    tokens_without_sw = [
        word for word in text_tokens if not word in STOP_WORDS]
    filtered_sentence = (" ").join(tokens_without_sw)
    text = filtered_sentence

    return text


def find_frequentwords(link):
    max_rows = 6000  # 'None' to read whole file
    input_file = link
    df = pd.read_csv(input_file,
                     delimiter = ',',
                     nrows = max_rows,
                     engine = "python")

    dt = df['comments'].apply(cleaning)

    word_count = Counter(" ".join(dt).split()).most_common(10)
    word_frequency = pd.DataFrame(word_count, columns = ['Word', 'Frequency'])
    print(word_frequency)

[nltk_data] Downloading package punkt to
[nltk_data]     C:\Users\lwl_1\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\lwl_1\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


In [52]:
print("Goyard frequent words")
find_frequentwords("goyard.csv")

Goyard frequent words
           Word  Frequency
0          love         50
1          need         33
2        wallet         33
3           bag         29
4           new         29
5        goyard         21
6         louis         19
7  bagaholicboy         17
8            st         17
9          dont         14


In [66]:
print("Cartier frequent words")
find_frequentwords("cartier.csv")

Cartier frequent words
      Word  Frequency
0     love         77
1                 42
2               41
3  cartier         30
4                 23
5    bling         21
6                17
7      omg         16
8     nice         15
9  classic         13


In [67]:
print("Fendi frequent words")
find_frequentwords("fendi.csv")

Fendi frequent words
    Word  Frequency
0   cute         47
1               47
2   love         40
3  fendi         32
4             30
5    omg         23
6   need         23
7               20
8    bag         19
9              16


In [72]:
print("Celine frequent words")
find_frequentwords("celine.csv")

Celine frequent words
     Word  Frequency
0    love         41
1                41
2     bag         31
3              24
4    cute         23
5  celine         22
6    like         15
7               13
8                12
9     omg         12


In [73]:
print("Gucci frequent words")
find_frequentwords("gucci.csv")

Gucci frequent words
           Word  Frequency
0                      94
1          love         71
2          cute         69
3         gucci         46
4                      42
5                    41
6           omg         31
7          like         30
8  bagaholicboy         29
9                    28


In [74]:
print("Hermes frequent words")
find_frequentwords("hermes.csv")

Hermes frequent words
           Word  Frequency
0                      93
1          love         65
2                    60
3                      49
4          need         41
5         black         34
6           omg         31
7           bag         30
8  bagaholicboy         30
9         kelly         29


In [82]:
print("Prada frequent words")
find_frequentwords("prada.csv")

Prada frequent words
    Word  Frequency
0               39
1   love         32
2             29
3   cute         24
4              19
5  prada         19
6    bag         18
7               17
8  bling         17
9             15


In [83]:
print("LV frequent words")
find_frequentwords("lv.csv")

LV frequent words
           Word  Frequency
0                     243
1          love        175
2          cute        126
3                   122
4                     102
5           bag        101
6           omg         72
7                     69
8          like         67
9  bagaholicboy         63


In [84]:
print("Dior frequent words")
find_frequentwords("dior.csv")

Dior frequent words
           Word  Frequency
0                     339
1          love        264
2                   204
3          dior        162
4                     126
5           omg        111
6                    109
7        please         96
8           bag         95
9  bagaholicboy         82


In [85]:
print("Chanel frequent words")
find_frequentwords("chanel.csv")

Chanel frequent words
           Word  Frequency
0                     517
1          love        430
2                   398
3                     222
4          cute        216
5        chanel        211
6           omg        203
7           bag        187
8                    183
9  bagaholicboy        164


#find interesting comments

#Common words that people relate to the brands
#TOP 1
Chanel
--> love (430), cute (216), chanel (211), omg (203), bag (187)

#TOP 2
Dior
--> love (264), dior (162), omg (111), please (96), bag (95)

#TOP 3
Louis Vuitton
-->  love (175), cute (126), bag (101), omg (72), like (67)

#TOP 4
Prada
--> love (32), cute (24), prada (19), bag (18), bling (17)

#TOP 5
Hermes
--> love (65), need (41), black (34), omg (31), bag (30)

#TOP 6
Gucci
--> love (71), cute (69), gucci (49), omg (31), like (30)

#TOP 7
Celine
--> Love (41), bag (31), cute (23), celine (22), like (15)

#TOP 8
Fendi
--> Cute (47), love (40), fendi (32), omg (30), need (23)

#TOP 9
Cartier
--> Love (77), Cartier (30), Bling (27), omg (17), nice (15)

#TOP 10
Goyard
--> Love (50), need (33), wallet (33), bag (29), new (29)